### ISSUE TITLE
`zfs mount` reports success but does not mount encrypted datasets

### ISSUE THREAD

**Issue by DHowett** [openzfs/zfs#13588]
### System information

Type | Version/Name
 --- | ---
Distribution Name	| Debian
Distribution Version	| sid
Kernel Version	| 5.18.5
Architecture	| amd64
OpenZFS Version	| 2.1.5-1

### Describe the problem you're observing

I have an encrypted dataset, `acentauri/vault`, with a loaded key.

```
# zfs get keystatus,mountpoint acentauri/vault
NAME             PROPERTY    VALUE                 SOURCE
acentauri/vault  keystatus   available             -
acentauri/vault  mountpoint  /mnt/acentauri/vault  inherited from acentauri
```

When I mount it, `zfs mount` reports success but nothing is mounted.

```
# mount | grep vault
# zfs mount acentauri/vault; echo $?
0
# ls /mnt/acentauri/vault
# mount | grep vault
#
```

No errors are reported in `dmesg`, no events are reported in `zfs events`, and `zfs mount -v` does not generate any additional information.

This regressed when I moved from 2.1.4 to 2.1.5, and from 5.17 to 5.18. I'll test out 5.17 x 2.1.5 and 5.18 x 2.1.4 shortly.

### Describe how to reproduce the problem

I am not certain; have an encrypted dataset and attempt to mount it on 2.1.5.

### Include any warning/errors/backtraces from the system logs

No logs are produced for this issue.

**Comment by DHowett** [comment]
Additional information. Rolling back through old versions of the module and the kernel reveals some surprising behavior: it no longer works on 2.1.4 as far back as 5.16.

| Module Version | Kernel Version | Outcome
| - | - | -
| 2.1.4 | 5.16 | ❌ mount is not populated
| 2.1.4 | 5.17 | ❌ mount is not populated
| 2.1.5 | 5.17 | ❌ mount is not populated
| 2.1.5 | 5.18 | ❌ mount is not populated

**If I set the mount point explicitly,** it works.

```
# zfs mount -v acentauri/vault
# mount | grep vault
... nothing ...

# zfs set mountpoint=/mnt/a acentauri/vault
# zfs mount -v acentauri/vault
# mount | grep vault
acentauri/vault on /mnt/a type zfs (rw,relatime,xattr,posixacl)
# ls /mnt/a | head -n 1
filename.ext
```

**Comment by mtippmann** [comment]
see also https://github.com/openzfs/zfs/issues/12418 - problem is still there with latest git - checked a few days ago

**Comment by DHowett** [comment]
Ah, apparently I should have gone beyond the first two pages of search results when I tried to find an existing report of this. Thanks! This one is definitely a duplicate.

**Comment by devZer0** [comment]
https://zfsonlinux.topicbox.com/groups/zfs-discuss/Tfeca5c3e85c35c8a/help-recovering-pool-zfs-ate-all-my-files

**Comment by devZer0** [comment]
does anybody still experience this problem ? @DHowett ? @mtippmann ?

@

**Comment by DHowett** [comment]
Yes. This issue has not miraculously fixed itself in the way the "stale" robot would like.
